Development/ssl/ssl_utils.h

```diff
@@ -0,0 +1,46 @@
+#ifndef SSL_SSL_UTILS_H
+#define SSL_SSL_UTILS_H
+
+#include <memory>
+#include <stdexcept>
+#include <string>
+#include <vector>
+#include <openssl/x509v3.h> // for X509V3_EXT_conf_nid
+
+namespace ssl
+{
+    namespace experimental
+    {
+        struct ssl_exception : std::runtime_error
+        {
+            ssl_exception(const std::string& message) : std::runtime_error(message) {}
+        };
+
+        typedef std::unique_ptr<BIO, decltype(&BIO_free)> BIO_ptr;
+        typedef std::unique_ptr<X509, decltype(&X509_free)> X509_ptr;
+        typedef std::unique_ptr<GENERAL_NAMES, decltype(&GENERAL_NAMES_free)> GENERAL_NAMES_ptr;
+        typedef std::unique_ptr<ASN1_TIME, decltype(&ASN1_STRING_free)> ASN1_TIME_ptr;
+
+        // get last openssl error
+        std::string last_openssl_error();
+
+        struct cert_info
+        {
+            std::string common_name;
+            std::string issuer_name;
+            time_t not_before;
+            time_t not_after;
+            std::vector<std::string> sans;
```

Comment:
Should we spell out `subject_alternative_names` if we're spelling out `common_name`?